### PR TITLE
Lessons widget

### PR DESCRIPTION
**Pull request by MarekMacko**

### CODE REVIEW

File: src/main/java/pl/edu/zut/mad/schedule/data/ScheduleRepository.kt [hunk 1]
```diff
@@ -26,15 +26,14 @@ internal class ScheduleRepository(private val database: ScheduleDatabase, privat
...
-    fun getDayByDate(date: LocalDate): Observable<OptionalDay> =
-        Observable.fromCallable<OptionalDay> {
-            database.instance.where(DayApi::class.java)
-                .equalTo(DATE_COLUMN, date.toDate())
-                .findFirst()
-                ?.asFlowable<DayApi>()
-                ?.map { mapper.toDayUiFromApi(it) }
-                ?.blockingFirst() ?: EmptyDay(date)
-        }
+    fun getDayByDate(date: LocalDate): Observable<OptionalDay> {
```
Comment (sswierczek) [inline src/main/java/pl/edu/zut/mad/schedule/data/ScheduleRepository.kt:29]: Is this tested?

File: src/main/java/pl/edu/zut/mad/schedule/util/ModelMapper.kt [hunk 2]
```diff
@@ -29,10 +30,14 @@ internal class ModelMapper {
                 val subjectWithCourseType = "$subject ($courseType)"
                 val teacherFullNameWithRoom = "${teacher?.academicTitle} ${teacher?.name} ${teacher?.surname} $room"
                 val isCancelled = reservationStatus.equals(CANCELED_LESSON_TEXT, true)
-                LessonUi(teacherFullNameWithRoom, subjectWithCourseType, isCancelled, timeRange ?: TimeRange())
+                val timeRangeUi = toUiTimeRange(timeRange ?: TimeRangeApi())
```
Comment (sswierczek) [inline src/main/java/pl/edu/zut/mad/schedule/util/ModelMapper.kt:33]: Is this tested?

File: src/main/res/layout/widget_lessons.xml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        android:background="@color/scheduleColorPrimary">
+
+        <TextView
+            android:id="@+id/widgetDateView"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:layout_gravity="center"
+            android:gravity="center"
+            android:textColor="@color/white"
+            android:textSize="16sp"
```
Comment (sswierczek) [inline src/main/res/layout/widget_lessons.xml]: dimens.xml

**Review by sswierczek — CHANGES_REQUESTED** [review]

File: src/main/java/pl/edu/zut/mad/schedule/util/ModelMapper.kt (hunk 2 above)
Comment (MarekMacko) [reply src/main/java/pl/edu/zut/mad/schedule/util/ModelMapper.kt:33]: Tests already exists, but they have to been fixed with  https://github.com/MADWI/schedule/pull/7/commits/2b383b96196b9ea5f327f9eceba3c208c6660c53 :)

File: src/main/java/pl/edu/zut/mad/schedule/data/ScheduleRepository.kt (hunk 1 above)
Comment (MarekMacko) [reply src/main/java/pl/edu/zut/mad/schedule/data/ScheduleRepository.kt:29]: Please, check https://github.com/MADWI/schedule/pull/7/commits/66b8c58de98ec4d767bbedb68a0bc8d16f6bc1ce :)

**Review by sswierczek — APPROVED** [review]

**Pull request merged**